harmoni-web-api/src/main/java/org/harmoniapp/harmoniwebapi/controllers/MessageController.java

```diff
@@ -0,0 +1,46 @@
+package org.harmoniapp.harmoniwebapi.controllers;
+
+import lombok.RequiredArgsConstructor;
+import org.harmoniapp.harmoniwebapi.contracts.MessageDto;
+import org.harmoniapp.harmoniwebapi.services.MessageService;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("message")
+@CrossOrigin(origins = "http://localhost:3000")
+public class MessageController {
+    private final MessageService service;
+
+//    @GetMapping("/history")
+//    public List<MessageDto> getChatHistory(@RequestParam Long userId1, @RequestParam Long userId2) {
+//        return service.getChatHistory(userId1, userId2);
+//    }
+
+    @GetMapping("/history")
+    public List<MessageDto> getChatHistory(
+            @RequestParam Long userId1,
+            @RequestParam Long userId2,
+            @RequestParam(defaultValue = "false") boolean translate,
+            @RequestParam(required = false) String targetLanguage) {
+        return service.getChatHistory(userId1, userId2, translate, targetLanguage);
+    }
```

Comment:
Chat should be with pagination or other division such as date range. Currently, it is loaded in its entirety, which will eventually heavily increase the processing time of the query in case of a large number of messages.